### PR TITLE
Impulse commits

### PR DESCRIPTION
**Pull request by Vectorphy**
Added features by impulse. Creating a PR for same.

### CODE REVIEW

**Review by 7han0s — CHANGES_REQUESTED** [review]
Overall, the changes look solid, and we should be ready to merge these commits. While I've done a detailed review, I haven’t had the chance to write everything down yet (some of the key points were discussed on Discord). Please bear with me as I outline the immediate issues.

_Ideally, I’ll add more detailed comments directly to the relevant sections of the code._

### Immediate Priorities:

1. **Merge Risk & Repository Integrity**:
   - ~~The main goal with this merge is to check for any issues with the repository, specifically in terms of file consistency. We’re possibly reintroducing files from **@ImpulseADHD’s branch** that were dropped by **@Vectorphy** in the main branch. Hopefully this won't undo the groundwork for unit tests, automations and CI&CD. In theory, this shouldn’t cause any conflicts or breaks unless changes were made to the dropped files.~~
   
2. **Action Items**:
   - **@ImpulseADHD**: If you have any additional commits relevant to this PR, please push them to your branch, and we’ll include them before merging. **Please remember to sync your branch after merges and changes to the main branch** 🥺
   - **@Vectorphy**: We need more clarity on what exactly broke after merging the last PR. Understanding the "why" and "how" will help us avoid repeating the same issues in this merge.

3. **Versioning**:
   - Let’s create **at least two releases** to maintain version control:
     - One before merging the last PR (v. Important).
     - One after merging this PR.
   - This gives us a safety net in case anything goes wrong, ensuring we have a clean rollback point.

4. **Codebase Alignment**:
   - We all need to get on the same page regarding the **codebase structure, repository setup, logic, conventions**, and project dependencies. Consistency is key to preventing further conflicts and simplifying merges.
   - Moving forward, we should all aim to make necessary changes in our individual branches in alignment with agreed-upon standards to avoid drastic conflicts.
---

**Comment by 7han0s** [comment]
## General Guidelines & Next Steps

### Best Practices:

1. **Coding Standards**:
   - Let’s ensure that we’re following **best practices** and maintaining code quality throughout the project.
   - Pay attention to **PEP8** compliance for readability and maintainability.
   - Discuss & Write **unit tests** to verify any new or updated functionality. These tests will help prevent bugs from slipping into production.

2. **Manual Testing**:
   - Even with automated tests, don’t forget to perform **manual sanity checks** (hehe) on areas that are difficult to test automatically (or where automation falls short).
   - This is especially important when introducing new features or functions. Let’s make sure nothing breaks what was working perfectly before.

### Deployment & Testing:

1. **Cloud Deployment**:
   - Once this PR is merged and validated, we should focus on **deploying the bot to the cloud**. There are several reasons why this is critical beyond what we've already discussed:
     - It will prevent issues where different team members have different local environments causing discrepancies.
     - A centralized environment will simplify testing and configurations, and reduce conflicts when making changes or running tests across multiple local instances.
   - With cloud deployment, we can focus on the core logic without worrying about local setup differences.

---

### Final Notes:

- **Kudos to the team** for making it this far. We’ve made great progress, but let’s continue tightening up the code and processes to ensure smoother collaboration and releases going forward.

- The suggested next steps before cloud deployment are:
   - Discuss and Validate the current codebase
   - List major issues, Fix minor issues
   - **Clean up** the repository and codebase: Remove obsolete code, dependencies, libraries and modules.
  
- Let’s stay aligned on these changes and practices to avoid introducing unnecessary complexity or risk into the project.

---

**Review by Vectorphy — COMMENTED** [review]
Overall DB looks way better now we just need to take all this and dump into mongoDB

File: utils.py [hunk 1]
```diff
@@ -107,8 +114,107 @@ def parse_custom_id(interaction: discord.Interaction) -> tuple:
...
+
+    except Exception as e:
+        logger.critical(f"Unexpected error during validation by user {interaction.user}: {e}")
+        await interaction.followup.send(f"An unexpected error occurred during validation: {e}", ephemeral=True)
+        return False
+
+
 
-### Assign Roles
+### Members and Roles Functions
```
Comment (Vectorphy) [inline utils.py]: Incomplete function?

**Pull request merged**